Apply proxy settings from vscode user configuration.

whyb opened the pull request:
Add proxy support.

common errors:

![圖片](https://github.com/16bit-ykiko/vscode-compiler-explorer/assets/2672931/e94e2228-7882-4cab-a374-12d941e36e70)

vscode http.proxy possibilities:
```
http://username:password@proxyserver:port
https://username:password@proxyserver:port
http://proxyserver:port
https://proxyserver:port
```

whyb committed:
Apply proxy settings from vscode user configuration.

16bit-ykiko commented:
Thanks for your contribution! I notice that there is some duplicated code. If you want to modify the logic of proxy, modify `SetProxy` function directly.

https://github.com/16bit-ykiko/vscode-compiler-explorer/blob/e292fec0a21a96d4e57790d51a70a4fc1b7c4d56/src/request/Utility.ts#L11-L20

whyb committed:
Apply proxy settings from vscode user configuration.

whyb commented:
> Thanks for your contribution! I notice that there is some duplicated code. If you want to modify the logic of proxy, modify `SetProxy` function directly.
> 
> https://github.com/16bit-ykiko/vscode-compiler-explorer/blob/e292fec0a21a96d4e57790d51a70a4fc1b7c4d56/src/request/Utility.ts#L11-L20

Sorry, I have never written in typescript language. If you have any questions, please point them out directly.

I have removed a lot of duplicate code as requested in the reply, and moved the function of setting axios proxy to the function SetProxy() in Utility.ts

16bit-ykiko commented:
It looks good for me. Thanks for your contribution!
